src/ServiceBusExplorer/Forms/CreateEventGridSubscriptionForm.cs

```
@@ -0,0 +1,356 @@
+                        filter["Value"] = Value;
+
+                        if (!string.IsNullOrEmpty(filter["Key"]))
+                        {
+                            filterList.Add(filter);
+                            writeToLog(string.Format(CultureInfo.CurrentCulture, $"Successfully added filter with key: {Key}, Operator: {Operator}, and Value(s): {Value}"));
+                        }
+
+
+                        if (!string.IsNullOrEmpty(EventType))
+                        {
+                            eventTypesList.Add(EventType);
+                            writeToLog(string.Format(CultureInfo.CurrentCulture, $"Successfully added EventType {EventType}"));
+                        }
+
+                        if (filterList.Count >= 25)
+                        {
+                            btnAddNewFilter.Enabled = false;
+                        }
+
+                        Key = String.Empty;
+                        Operator = String.Empty;
+                        Value = String.Empty;
+                        textBoxFilterKey.Text = String.Empty;
+                        comboBoxFilterOperator.Text = String.Empty;
+                        comboBoxFilterValue.Text = String.Empty;
+                        textBoxFilterEventType.Text = String.Empty;
+                        textBoxFilterValue.Text = String.Empty;
+                    }
+                }
+
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
+            }
+
+        }
+        private void btnCancel_Click(object sender, EventArgs e)
+        {
+            DialogResult = DialogResult.Cancel;
+            Close();
+        }
+
+        private void HandleException(Exception? ex)
+        {
+            if (string.IsNullOrWhiteSpace(ex?.Message))
+            {
+                return;
+            }
+            if (ex.GetType() == typeof(EventGridException))
+            {
+                throw ex;
```

Don't do: 
`throw ex;`
instead do:
`throw`

Otherwise the stack trace is lost.

---

He can't do that, @ErikMogensen. Not the way you suggested. The exception was not caught in this method with a try/catch. It was passed it.

But the problem is not rethrowing. Why do we even throw if the `HandleException()` method is supposed to log and swallow exceptions? What's so special about EventGrid that it requires to throw here? Isn't it going to crash the UI?

@harrieoriowo, could you answer these questions? Thank you.